test: Fix data-loaded handling and "stale element" failures in Selenium tests

martinpitt opened the pull request:
The `jscheck` parameter of the various wait*() function was not about a
"safer" way to wait for an element, but it just meant to wait for the
`data-loaded` attribute. This is only a thing for Cockpit's `<iframe>`s
though, not for any other element [1]. Thus any wait query that enabled
jscheck was just pointlessly retrying the wait 40 times.

Drop the argument from all specific wait_thing() functions, rename it to
`wait_data_loaded` for clarity, and only use it in wait_frame(). In
other words, now there is no reason to ever use this keyword in the
tests.

[1] except in the shell's credentials dialog for SSH keys, but there
it's not a transient loading indicator but a persistent state

martinpitt commented:
I added another commit that *should* fix these pesky "stale element reference" failures. Works fine  locally at least. @jsotka, can you please have a look?

jscotka commented on test/selenium/testlib_avocado/seleniumlib.py:
```diff
@@ -307,7 +293,7 @@ def select_by_text(self, element, value):
         return self.select(element=element, select_function="select_by_value", value=value)
 
-    def wait(self, method, text, baseelement, overridetry, fatal, cond, jscheck, text_):
+    def wait(self, method, text, baseelement, overridetry, fatal, cond, text_, wait_data_loaded=False):
```
I hope, that nobody uses positional parameters for this, but suggest to switch positions of ``text_, wait_data_loaded=False`` with this generic method, and no not pass default param, as this should be, but I understand that  the reason is to not pass this default to each specific functiono

martinpitt replied:
Ack, I changed it.

martinpitt commented:
@jscotka: I addressed your feedback and also fixed (hopefull) the [remaining two stale element failures](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13059-20191029-145252-0b99528b-cockpit-project-cockpit--fedora-30-selenium-chrome/log.html). Can you please have another look?

martinpitt commented:
@jscotka: Yay, now only "real" failures are left. I think this is good to go.
